### PR TITLE
chore: upgrade dbt version 1.0.3

### PR DESCRIPTION
**Pull request by jazibhumayun**
**Description:** Describe in a couple of sentences what this PR adds

**JIRA:** Link to JIRA ticket

**Dependencies:** dependencies on other outstanding PRs, issues, etc.

**Merge deadline:** List merge deadline (if any)

**Installation instructions:** List any non-trivial installation
instructions.

**Testing instructions:**

1. Open page A
2. Do thing B
3. Expect C to happen
4. If D happened instead - check failed.

**Reviewers:**
- [ ] tag reviewer
- [ ] tag reviewer

**Merge checklist:**
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Version bumped
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are squashed

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPI after tag-triggered build is
      finished.
- [ ] Delete working branch (if not needed anymore)

**Author concerns:** List any concerns about this PR - inelegant
solutions, hacks, quick-and-dirty implementations, concerns about
migrations, etc.

### CODE REVIEW

**Review by pwnage101 — APPROVED** [review]

File: requirements/base.in [hunk 1]
```diff
@@ -1,8 +1,9 @@
...
 -c constraints.txt
 
-dbt==0.21.1
+dbt-core==1.0.3
+dbt-snowflake==1.0.0
 
 # New release of MarkupSafe 2.1.0 is throwing an error "ImportError: cannot import name 'soft_unicode' from 'markupsafe'"
 # https://github.com/pallets/markupsafe/issues/282
-MarkupSafe==2.0.1
+# MarkupSafe==2.0.1
```
Comment (estute) [inline requirements/base.in]: Should we just remove this? Are we using it?

File: dbt_schema_builder/builder.py [hunk 2]
```diff
@@ -10,7 +10,8 @@
 import yaml
 from dbt.config import RuntimeConfig
 from dbt.exceptions import DatabaseException
-from dbt.logger import GLOBAL_LOGGER as logger
+from dbt.events import AdapterLogger
+logger = AdapterLogger("Snowflake")
```
Comment (estute) [inline dbt_schema_builder/builder.py]: nit: can you move this line down below? Keeping imports separate from actual code helps readability.

File: dbt_schema_builder/builder.py [hunk 3]
```diff
@@ -10,7 +10,8 @@
 import yaml
 from dbt.config import RuntimeConfig
 from dbt.exceptions import DatabaseException
-from dbt.logger import GLOBAL_LOGGER as logger
+from dbt.events import AdapterLogger
```
Comment (estute) [inline dbt_schema_builder/builder.py:12]: Do we still need the `log_manager` if you are setting the logger? Also, is the the logger variable used anywhere?

File: dbt_schema_builder/schema_builder.py [hunk 4]
```diff
@@ -73,6 +74,7 @@ def parse_args(args):
         sys.exit(1)
 
     parsed = p.parse_args(args)
+    flags.set_from_args(parsed, {})
```
Comment (estute) [inline dbt_schema_builder/schema_builder.py:77]: What happens to `flags`?

**Review by estute — COMMENTED** [review]
I have a couple of minor questions/nits. Aside from that, can you just explain a little about how this was tested, as the CI for this repo does not actually interface with Snowflake?

File: dbt_schema_builder/builder.py (hunk 3 above)
Comment (jazibhumayun) [reply dbt_schema_builder/builder.py:12]: `log_manager` is used here: [https://github.com/edx/dbt-schema-builder/blob/main/dbt_schema_builder/builder.py#L611](https://github.com/edx/dbt-schema-builder/blob/main/dbt_schema_builder/builder.py#L611)
but for rest of the places Snowflake AdapterLogger object `logger` is used.

File: dbt_schema_builder/schema_builder.py (hunk 4 above)
Comment (jazibhumayun) [reply dbt_schema_builder/schema_builder.py:77]: This was an issue with dbt cli command not picking the `--profile-dir` path, so this change is made to honor `--profile-dir` path

File: requirements/base.in (hunk 1 above)
Comment (jazibhumayun) [reply requirements/base.in]: Yes, we can remove this.

**Comment by jazibhumayun** [comment]
> I have a couple of minor questions/nits. Aside from that, can you just explain a little about how this was tested, as the CI for this repo does not actually interface with Snowflake?

Yes, we essentially follow these steps to upgrade and test dbt-schema-builder


- Update the dbt version here: https://github.com/edx/dbt-schema-builder/blob/master/requirements/base.in#L4
- Create a new Python 3.8 virtual environment using your favorite tool
- Run a make upgrade to update the versions in all of the local requirements files
- From the dbt-schema-builder directory run pip install . to get Schema Builder and dependencies installed to your virtualenv
- Perform a schema-builder PR-generation run by running the command below. 
- cd to the projects/automated/applications directory in warehouse-transforms
- Replace the --profiles-dir path then run this command: dbt_schema_builder build --destination-project ../../reporting --profile warehouse_transforms --target prod --profiles-dir /PATH/TO/analytics-secure/warehouse-transforms/ 
- Verify that the command successfully runs and updates only the expected files inside warehouse-transforms.
- Fix any errors caused by dbt changes in the new version.

File: dbt_schema_builder/schema_builder.py (hunk 4 above)
Comment (estute) [reply dbt_schema_builder/schema_builder.py:77]: I guess my question was more of: we are returning `parsed` but not `flags`. Are we leaving something out or does it create a side effect outside the scope of this function?

**Review by estute — APPROVED** [review]
One minor question, but LGTM

**Pull request merged**